lightningd/peer_control.c

```diff
@@ -1280,7 +1280,30 @@ static void handle_irrevocably_resolved(struct peer *peer, const u8 *msg)
  */
 static void onchain_add_utxo(struct peer *peer, const u8 *msg)
 {
-	/* FIXME(cdecker) Implement */
+	struct bitcoin_txid prev_out_tx;
+	u32 prev_out_index;
+	struct pubkey commitment_point;
+	u64 value;
+
+	if (!fromwire_onchain_add_utxo(msg, NULL, &prev_out_tx, &prev_out_index,
+				       &commitment_point, &value)) {
+		fatal("onchaind gave invalid add_utxo message: %s", tal_hex(msg, msg));
+	}
+
+	struct utxo *u = tal(msg, struct utxo);
+	u->is_p2sh = true;
+	u->txid = prev_out_tx;
+	u->outnum = prev_out_index;
+	u->keyindex = 0;
+	u->status = output_state_available;
+	u->amount = value;
```

Comment:
I would usually allocate first, initialize some fields, then directly pass fields to fromwire_onchain_add_utxo.  Minor though.